test-suite/Makefile.am

```diff
@@ -140,9 +141,41 @@ VirtualDeleteOperator_SOURCES = VirtualDeleteOperator.cc stub_libmem.cc $(DEBUG_
 ##$(TARGLIB): $(LIBOBJS)
 ##	$(AR_R) $(TARGLIB) $(LIBOBJS)
 
-squid-conf-tests: $(top_builddir)/src/squid.conf.default $(srcdir)/squidconf/*
-	@failed=0; cfglist="$?"; rm -f $@ || $(TRUE); \
-	for cfg in $$cfglist ; do \
+generate-test-certificates:
+	$(srcdir)/mk-tls-test-certs.sh "$(CERTTOOL_BIN)" "$(OPENSSL_BIN)"
+
+CLEANFILES += \
+	ca-chain-rsa.pem \
+	ca-mid-rsa.crt \
+	ca-mid-rsa.pem \
+	ca-mid-rsa.pkey \
+	ca-root-rsa.crt \
+	ca-root-rsa.pem \
+	ca-root-rsa.pkey \
+	leaf-chain-rsa.pem \
+	leaf-chain-nokey-rsa.pem \
+	leaf-key-rsa.pem \
+	leaf-rsa.crt \
+	leaf-rsa.pkey
```

Comment:
I do not insist on this change, but I think it would be better to name all these test files (that we are creating, overwriting, removing, etc.) starting with a common pattern that is less likely to clash with real files. For example, `test-ca-chain-rsa.pem` and such. Placing them in a temporary test directory would be even better, but I am guessing that it would be difficult.

Reply:
You expect other files with those names to exist in the Squid test-suite build directory? That would be highly inappropriate.

Reply:
> You expect other files with those names to exist in the Squid test-suite build directory? That would be highly inappropriate.

I assume that the "test-suite build directory" may be a directory in a working copy of a repository.  With that assumption in mind, personally, I see nothing particularly _inappropriate_ with a hypothetical developer placing something like `ca-root-rsa.pem` in the current directory which may happen to be the `test-suite/` directory of her repository. Perhaps she is working on a new custom test case, unaware that Squid itself generates and deletes that file.

The overall "do no harm" and "avoid surprises" principles tell us that temporary test files should either go into a temporary test directory or, at the very least, should be named in a way that is unlikely to clash with user files (and that makes it easy for humans to identify these files as temporary test files when tests fail to clean up).

BTW, if all these files were named appropriately, then we would not need to modify `CLEANFILES` when adding new similar files. Currently, our Makefile has to be in sync with the internal matters of some test cases. This information leakage is a sign of a test harness design problem.

However, I (still) do not insist on this change. I agree that the probability of a collision is low here, and I suspect it may require quite a bit of manual work to rename these files in this PR.

Reply:
Whatever name we choose runs the same risk of this case you are worried about. It also occurs with any file committed to the repository. This is why it is Bad Practice (inappropriate)  to build directly in the source directory structure. On the other hand developers are those we can most expect to be able to cope with the result of these clashes.

Reply:
> Whatever name we choose runs the same risk of this case you are worried about. It also occurs with any file committed to the repository.

I disagree on both counts, but I think it is best to agree to disagree here -- it is a minor issue that does not block this PR progress or acceptance and that involves non-technical judgement calls that make consensus especially difficult to reach.

Reply:
Resolving this conversation since we agree to disagree.